File: src/lib/playingRoles.ts
import type { PlayingRole } from '../types'

export const PLAYING_ROLE_LABELS: Record<PlayingRole, string> = {
  batsman: 'Batsman',
  bowler: 'Bowler',
  allRounder: 'All-rounder',
  battingAllRounder: 'Batting All Rounder',
  bowlingAllRounder: 'Bowling All Rounder',
  legSpinner: 'Leg Spinner',
  offSpinner: 'Off Spinner',
  legOffSpinner: 'Leg-Off Spinner',
  wicketKeeper: 'Wicket-keeper',
}

export const PLAYING_ROLE_OPTIONS: { value: PlayingRole; label: string }[] = (
  Object.entries(PLAYING_ROLE_LABELS) as [PlayingRole, string][]
).map(([value, label]) => ({ value, label }))

File: android/app/src/main/java/com/mplauction/android/data/PlayingRoles.kt
package com.mplauction.android.data

import com.mplauction.android.data.model.PlayingRole

// Mirrors PLAYING_ROLE_LABELS in lib/playingRoles.ts — used to snapshot a
// registered player's profile role as their auction lot's position.
val PLAYING_ROLE_LABELS =
  mapOf(
    PlayingRole.batsman to "Batsmen",
    PlayingRole.bowler to "Bowler",
    PlayingRole.battingAllRounder to "Batting All Rounder",
    PlayingRole.bowlingAllRounder to "Bowling All Rounder",
    PlayingRole.wicketKeeperBatsman to "Wicket Keeper Batsmen",
  )
